src/pipelinedefinition.cpp

```diff
@@ -492,33 +492,34 @@ class NodeValidator {
             return StatusCode::PIPELINE_NOT_ENOUGH_SHAPE_DIMENSIONS_TO_DEMULTIPLY;
         }
         if (demultiplicatorNodeInfo.demultiplyCount.value() != 0) {
+            // TODO -1 vs 0 as demultiplexing value
             if (shape[0] != 0) {
-                // 0 means that node accepts dynamic shape
-                if (shape[0] != demultiplicatorNodeInfo.demultiplyCount.value()) {
+                auto demultiplyDimension = demultiplicatorNodeInfo.demultiplyCount.value() ? Dimension(demultiplicatorNodeInfo.demultiplyCount.value()) : Dimension::any();
+                if (!shape[0].fitsInto(demultiplyDimension)) {
                     SPDLOG_LOGGER_ERROR(modelmanager_logger, "Validation of pipeline: {} definition failed. Demultiply count: {} of node: {} does not match tensor first dimension value: {}",
                         this->pipelineName,
                         demultiplicatorNodeInfo.demultiplyCount.value(),
                         demultiplicatorNodeInfo.nodeName,
-                        shape[0]);
+                        shape[0].toString());
                     return StatusCode::PIPELINE_DEMULTIPLY_COUNT_DOES_NOT_MATCH_BLOB_SHARD_COUNT;
                 }
             } else {
                 SPDLOG_LOGGER_WARN(modelmanager_logger, "Demultiply count: {} of node: {} is fixed while first dimenson value of node library is not: {}. This pipeline may fail at execution stage.",
                     demultiplicatorNodeInfo.demultiplyCount.value(),
                     demultiplicatorNodeInfo.nodeName,
-                    shape[0]);
+                    shape[0].toString());
             }
-        } else if (shape[0] != 0) {
-            SPDLOG_LOGGER_WARN(modelmanager_logger, "Demultiply count: {} of node: {} is dynamic while first dimenson value of gather node is not: {}. This pipeline may fail at execution stage.",
+        } else if (!shape[0].isAny() && shape[0] != 0) {
```

Comment:
Here the `shape[0] != 0` is wrong, since shape[0] being equal to 0 does not mean this is any. 0 is static 0